csharp/src/Ice/Communicator.cs

```diff
@@ -759,28 +757,27 @@ internal CancellationToken CancellationToken
             }
         }
 
-        // TODO: temporarily private
-        private async Task InitializeAsync(CancellationToken cancel = default)
+        /// <summary>Activates the plugins and built-in locator implementation of this communicator, if any. Also
+        /// creates and activates the Ice.Admin object and its object adapter, if enabled. It is recommended to always
+        /// activate a communicator even though this activation may not do anything.</summary>
+        /// <param name="cancel">The cancellation token.</param>
+        /// <returns>A task that completes when the activation completes.</returns>
+        public async Task ActivateAsync(CancellationToken cancel = default)
         {
-            // An application can set Ice.InitPlugins=0 if it wants to postpone initialization until after it has
-            // interacted directly with the plug-ins.
-            if (GetPropertyAsBool("Ice.InitPlugins") ?? true)
+            if (_activateCalled)
             {
-                await InitializePluginsAsync(cancel).ConfigureAwait(false);
+                throw new InvalidOperationException("ActivateAsync was already called on this communicator");
             }
+            _activateCalled = true;
```

Comment:
Why don't use the lock to set _activateCalled, this doesn't seem thread safe

Reply:
You're right, this is incorrect. Fixed.